### PR TITLE
PN-11342-adattare-la-solution-di-ta-fe-alle-modifiche-introdotte-dal-pn-bff

### PR DESCRIPTION
**Pull request by afus0**
## Short description

This pull request is still in draft.

## List of changes proposed in this pull request

- added  `setApiBaseUrl()` method in order to set up the right environment for the API
- added `apiBaseUrl` property into `test-config.properties`
- added `downloadFolder` property into `test-config.properties`
- added experimental options in chrome
- added method `setUpDownloadFolder(), removeDownloadFolder()` in order to remove the folder created

##

- [x] Does this PR require a manual test in AWS CodeBuild?

### CODE REVIEW

**Comment by leleOFA** [comment]
during a call with Emanuele Bonofiglio, you should take all new url in one application properties

File: src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java [hunk 1]
```diff
@@ -174,4 +174,10 @@ public void selezioneAvvisoPagoPa() {
             Assert.fail("check su avviso pagopa non avvenuto con successo: " + e.getMessage());
         }
     }
+
+    public void clickFile(String fileName) {
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/common/DettaglioNotificaSection.java]: explain better this method, please

File: src/main/java/it/pn/frontend/e2e/listeners/Hooks.java [hunk 2]
```diff
@@ -84,8 +85,16 @@ protected void chrome() {
 
         chromeOptions.setExperimentalOption("prefs", chromePrefs);
         if (this.headless != null && this.headless.equalsIgnoreCase("true")) {
+            downloadFilepath = Path.of("").toAbsolutePath().toString() + System.getProperty("downloadFolder").replace("/", File.separator);
```
Comment (leleOFA) [inline src/main/java/it/pn/frontend/e2e/listeners/Hooks.java]: this code It was already added by me

File: src/test/java/it/pn/frontend/e2e/run/RunCucumberTest.java [hunk 3]
```diff
@@ -49,8 +54,62 @@ private static boolean loadProperties() {
         return false;
     }
 
+    private static void setApiBaseUrl() {
+        String baseUrlProperty = System.getProperty("apiBaseUrl");
+        String env = System.getProperty("environment");
+        if (baseUrlProperty == null || env == null) {
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/run/RunCucumberTest.java]: I would insert an assert

File: src/test/java/it/pn/frontend/e2e/run/RunCucumberTest.java [hunk 4]
```diff
@@ -49,8 +54,62 @@ private static boolean loadProperties() {
...
+    }
+
+    private static void setUpDownloadFolder() {
+        String downloadDir = System.getProperty("downloadFolder");
+        Path fullPath = Paths.get(System.getProperty("user.dir"), downloadDir.replace("/", File.separator));
+        if (Files.notExists(fullPath)) {
+            try {
+                Files.createDirectories(fullPath);
+                logger.info("Download directory created: " + fullPath.toAbsolutePath());
+            } catch (IOException e) {
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/run/RunCucumberTest.java]: as before

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/LoginPersonaFisicaPagoPA.java [hunk 5]
```diff
@@ -49,11 +48,12 @@ public void loginPageDestinatarioVieneVisualizzata(String datipersonaFisica) {
...
     }
-        @Given("Login Page persona fisica test viene visualizzata")
-        public void loginPageDestinatarioVieneVisualizzataConUrl() {
 
-           String url = "https://cittadini.test.notifichedigitali.it/";
-            this.driver.get(url);
+    @Given("Login Page persona fisica test viene visualizzata")
+    public void loginPageDestinatarioVieneVisualizzataConUrl() {
+
+        String url = "https://cittadini.test.notifichedigitali.it/";
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/LoginPersonaFisicaPagoPA.java:55]: view https://github.com/pagopa/pn-frontend-e2e/pull/176 and proceed similarly

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java [hunk 6]
```diff
@@ -282,22 +278,22 @@ public void downloadFileAAR(){
...
-            int count = 1;
+    public void downloadAttestazioniOpponibili() {
+        logger.info("Si scaricano solo i file per attestazioni opponibili all'interno della notifica");
+
+        DettaglioNotificaMittenteSection dettaglioNotificaMittenteSection = new DettaglioNotificaMittenteSection(this.driver);
+        DettaglioNotificaSection dettaglioNotificaSection = new DettaglioNotificaSection(this.driver);
+        DataPopulation dataPopulation = new DataPopulation();
+
+        String workingDirectory = System.getProperty("user.dir");
+        String variabileAmbiente = System.getProperty("environment");
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java]: same as before comment

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java [hunk 7]
```diff
@@ -358,15 +353,11 @@ public void siSelezionanoIlFileDaScaricare(String nomeFile) {
...
-            TimeUnit.SECONDS.sleep(5);
-        } catch (InterruptedException e) {
-            throw new RuntimeException(e);
-        }
+        WebTool.waitTime(5);
         downloadFile = new DownloadFile(this.driver);
 
-        String variabileAmbiente = System.getProperty("environment");
-        final String url = downloadFile.getUrl("https://webapi." + variabileAmbiente + ".notifichedigitali.it/delivery-push/");
+        String varabileAmbiente = System.getProperty("environment");
```
Comment (leleOFA) [inline src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java]: correct in variabile

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/LoginPersonaFisicaPagoPA.java (hunk 5 above)
Comment (afus0) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/destinatario/personaFisica/LoginPersonaFisicaPagoPA.java:55]: i didn't change anything about this part, just formatted

File: src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java (hunk 7 above)
Comment (afus0) [reply src/test/java/it/pn/frontend/e2e/stepDefinitions/mittente/DownloadFileMittentePagoPATest.java]: removed because unused

**Review by leleOFA — APPROVED** [review]

**Pull request merged**